chromadb/api/segment.py

```diff
@@ -110,6 +111,24 @@ def __init__(self, system: System):
     def heartbeat(self) -> int:
         return int(time.time_ns())
 
+    @override
+    def optimize(self) -> OptimizationStats:
+        # loop through segments without loading them only the pickle file, as a fallback we can load the segment - we need max_seq_id
+        # call optimize on sysdb (for single-node this works just fine)
+        wal_entries_purged = 0
+        for s in self._sysdb.get_segments(scope=t.SegmentScope.VECTOR):
+            mtr = self._manager.get_segment(s["collection"], VectorReader)  # type: ignore
+            wal_entries_purged += self._producer.optimize(s["topic"])  # type: ignore
+            if s["collection"] not in self._collection_cache:
```

Comment:
@HammadB, This is iffy as we don't hold a lock over the collection, and if a request comes in during the execution of `optimize` we can end up in a race condition